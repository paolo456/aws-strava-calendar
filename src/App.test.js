import Container, {callRefreshURL} from './App.js'
import React from 'react'
import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

Enzyme.configure({ adapter: new Adapter() });

it('Open container and check sidebar and calendar', () => {
	const wrapper = Enzyme.mount(<Container></Container>);
	
	const container = wrapper.find('.container')
	expect(container).toBeTruthy()

	const listToggle = wrapper.find('.list-detail-toggle').text()
	expect(listToggle).toEqual('List View')

	const defaultActivity = wrapper.find('.default-activity-number').text()
	expect(defaultActivity).toEqual('20 Activities')

	const allActivites = wrapper.find('.all-activities').text()
	expect(allActivites).toEqual('All Activities')

	const delayButton = wrapper.find('.delay-button').text()
	expect(delayButton).toEqual('Send Delay')

	const calendar = wrapper.find('.calendar').text()
	expect(calendar).toBeTruthy()
});
