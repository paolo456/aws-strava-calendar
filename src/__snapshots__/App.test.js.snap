// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="loading"
      href="#"
    >
      <img
        alt=""
        height="400"
        src="test-file-stub"
        width="400"
      />
    </div>
    <div
      class="maps"
    >
      <div
        class="flex-box"
      />
    </div>
    <div
      class="filter"
    >
      <ul
        id="myUL"
      >
        <li>
          <a>
            List View
          </a>
        </li>
        <li>
          <a>
            20 Activities
          </a>
        </li>
        <li>
          <a>
            All Activities
          </a>
        </li>
        <li />
        <li>
          <form
            class="delay-form"
          >
            <input
              class="delay-input"
              name="delay"
              placeholder="Delay in ms"
              type="text"
            />
            <button
              class="delay-button"
            >
              Send Delay
            </button>
          </form>
        </li>
      </ul>
    </div>
    <div
      class="calendar"
    >
      <div>
        <div
          class="react-datepicker"
        >
          <div
            class="react-datepicker__triangle"
          />
          <button
            aria-label="Previous Month"
            class="react-datepicker__navigation react-datepicker__navigation--previous"
            type="button"
          >
            Previous Month
          </button>
          <button
            aria-label="Next Month"
            class="react-datepicker__navigation react-datepicker__navigation--next"
            type="button"
          >
            Next Month
          </button>
          <div
            class="react-datepicker__month-container"
          >
            <div
              class="react-datepicker__header "
            >
              <div
                class="react-datepicker__current-month"
              >
                March 2021
              </div>
              <div
                class="react-datepicker__header__dropdown react-datepicker__header__dropdown--scroll"
              />
              <div
                class="react-datepicker__day-names"
              >
                <div
                  class="react-datepicker__day-name"
                >
                  Su
                </div>
                <div
                  class="react-datepicker__day-name"
                >
                  Mo
                </div>
                <div
                  class="react-datepicker__day-name"
                >
                  Tu
                </div>
                <div
                  class="react-datepicker__day-name"
                >
                  We
                </div>
                <div
                  class="react-datepicker__day-name"
                >
                  Th
                </div>
                <div
                  class="react-datepicker__day-name"
                >
                  Fr
                </div>
                <div
                  class="react-datepicker__day-name"
                >
                  Sa
                </div>
              </div>
            </div>
            <div
              aria-label="month  2021-03"
              class="react-datepicker__month"
            >
              <div
                class="react-datepicker__week"
              >
                <div
                  aria-disabled="false"
                  aria-label="Choose Sunday, February 28th, 2021"
                  class="react-datepicker__day react-datepicker__day--028 react-datepicker__day--weekend react-datepicker__day--outside-month"
                  role="button"
                  tabindex="-1"
                >
                  28
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Monday, March 1st, 2021"
                  class="react-datepicker__day react-datepicker__day--001"
                  role="button"
                  tabindex="-1"
                >
                  1
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Tuesday, March 2nd, 2021"
                  class="react-datepicker__day react-datepicker__day--002"
                  role="button"
                  tabindex="-1"
                >
                  2
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Wednesday, March 3rd, 2021"
                  class="react-datepicker__day react-datepicker__day--003"
                  role="button"
                  tabindex="-1"
                >
                  3
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Thursday, March 4th, 2021"
                  class="react-datepicker__day react-datepicker__day--004"
                  role="button"
                  tabindex="-1"
                >
                  4
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Friday, March 5th, 2021"
                  class="react-datepicker__day react-datepicker__day--005"
                  role="button"
                  tabindex="-1"
                >
                  5
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Saturday, March 6th, 2021"
                  class="react-datepicker__day react-datepicker__day--006 react-datepicker__day--weekend"
                  role="button"
                  tabindex="-1"
                >
                  6
                </div>
              </div>
              <div
                class="react-datepicker__week"
              >
                <div
                  aria-disabled="false"
                  aria-label="Choose Sunday, March 7th, 2021"
                  class="react-datepicker__day react-datepicker__day--007 react-datepicker__day--weekend"
                  role="button"
                  tabindex="-1"
                >
                  7
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Monday, March 8th, 2021"
                  class="react-datepicker__day react-datepicker__day--008"
                  role="button"
                  tabindex="-1"
                >
                  8
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Tuesday, March 9th, 2021"
                  class="react-datepicker__day react-datepicker__day--009"
                  role="button"
                  tabindex="-1"
                >
                  9
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Wednesday, March 10th, 2021"
                  class="react-datepicker__day react-datepicker__day--010"
                  role="button"
                  tabindex="-1"
                >
                  10
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Thursday, March 11th, 2021"
                  class="react-datepicker__day react-datepicker__day--011"
                  role="button"
                  tabindex="-1"
                >
                  11
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Friday, March 12th, 2021"
                  class="react-datepicker__day react-datepicker__day--012"
                  role="button"
                  tabindex="-1"
                >
                  12
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Saturday, March 13th, 2021"
                  class="react-datepicker__day react-datepicker__day--013 react-datepicker__day--weekend"
                  role="button"
                  tabindex="-1"
                >
                  13
                </div>
              </div>
              <div
                class="react-datepicker__week"
              >
                <div
                  aria-disabled="false"
                  aria-label="Choose Sunday, March 14th, 2021"
                  class="react-datepicker__day react-datepicker__day--014 react-datepicker__day--weekend"
                  role="button"
                  tabindex="-1"
                >
                  14
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Monday, March 15th, 2021"
                  class="react-datepicker__day react-datepicker__day--015"
                  role="button"
                  tabindex="-1"
                >
                  15
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Tuesday, March 16th, 2021"
                  class="react-datepicker__day react-datepicker__day--016"
                  role="button"
                  tabindex="-1"
                >
                  16
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Wednesday, March 17th, 2021"
                  class="react-datepicker__day react-datepicker__day--017"
                  role="button"
                  tabindex="-1"
                >
                  17
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Thursday, March 18th, 2021"
                  class="react-datepicker__day react-datepicker__day--018"
                  role="button"
                  tabindex="-1"
                >
                  18
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Friday, March 19th, 2021"
                  class="react-datepicker__day react-datepicker__day--019"
                  role="button"
                  tabindex="-1"
                >
                  19
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Saturday, March 20th, 2021"
                  class="react-datepicker__day react-datepicker__day--020 react-datepicker__day--weekend"
                  role="button"
                  tabindex="-1"
                >
                  20
                </div>
              </div>
              <div
                class="react-datepicker__week"
              >
                <div
                  aria-disabled="false"
                  aria-label="Choose Sunday, March 21st, 2021"
                  class="react-datepicker__day react-datepicker__day--021 react-datepicker__day--weekend"
                  role="button"
                  tabindex="-1"
                >
                  21
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Monday, March 22nd, 2021"
                  class="react-datepicker__day react-datepicker__day--022"
                  role="button"
                  tabindex="-1"
                >
                  22
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Tuesday, March 23rd, 2021"
                  class="react-datepicker__day react-datepicker__day--023 react-datepicker__day--selected react-datepicker__day--range-start react-datepicker__day--range-end react-datepicker__day--in-range react-datepicker__day--today"
                  role="button"
                  tabindex="0"
                >
                  23
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Wednesday, March 24th, 2021"
                  class="react-datepicker__day react-datepicker__day--024"
                  role="button"
                  tabindex="-1"
                >
                  24
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Thursday, March 25th, 2021"
                  class="react-datepicker__day react-datepicker__day--025"
                  role="button"
                  tabindex="-1"
                >
                  25
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Friday, March 26th, 2021"
                  class="react-datepicker__day react-datepicker__day--026"
                  role="button"
                  tabindex="-1"
                >
                  26
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Saturday, March 27th, 2021"
                  class="react-datepicker__day react-datepicker__day--027 react-datepicker__day--weekend"
                  role="button"
                  tabindex="-1"
                >
                  27
                </div>
              </div>
              <div
                class="react-datepicker__week"
              >
                <div
                  aria-disabled="false"
                  aria-label="Choose Sunday, March 28th, 2021"
                  class="react-datepicker__day react-datepicker__day--028 react-datepicker__day--weekend"
                  role="button"
                  tabindex="-1"
                >
                  28
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Monday, March 29th, 2021"
                  class="react-datepicker__day react-datepicker__day--029"
                  role="button"
                  tabindex="-1"
                >
                  29
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Tuesday, March 30th, 2021"
                  class="react-datepicker__day react-datepicker__day--030"
                  role="button"
                  tabindex="-1"
                >
                  30
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Wednesday, March 31st, 2021"
                  class="react-datepicker__day react-datepicker__day--031"
                  role="button"
                  tabindex="-1"
                >
                  31
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Thursday, April 1st, 2021"
                  class="react-datepicker__day react-datepicker__day--001 react-datepicker__day--outside-month"
                  role="button"
                  tabindex="-1"
                >
                  1
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Friday, April 2nd, 2021"
                  class="react-datepicker__day react-datepicker__day--002 react-datepicker__day--outside-month"
                  role="button"
                  tabindex="-1"
                >
                  2
                </div>
                <div
                  aria-disabled="false"
                  aria-label="Choose Saturday, April 3rd, 2021"
                  class="react-datepicker__day react-datepicker__day--003 react-datepicker__day--weekend react-datepicker__day--outside-month"
                  role="button"
                  tabindex="-1"
                >
                  3
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
